spec/cornerstone/SEOAssessorSpec.js

```diff
@@ -1,32 +1,159 @@
 		assessor.assess( new Paper( "" ) );
-		expect( assessor.getValidResults().length ).toBe( 4 );
-	});
+		let assessments = getResults( assessor.getValidResults() );
 
-	it( "additionally runs assessments that only require a text", function() {
+		expect( assessments ).toEqual( [
+			"keyphraseLength",
+			"metaDescriptionLength",
+			"textLength",
+			"titleWidth"
+		] );
+	} );
+
+	it( "runs assessments that only require a keyword", function() {
+		assessor.assess( new Paper( "", { keyword: "keyword" } ) );
+		let assessments = getResults( assessor.getValidResults() );
+
+		expect( assessments ).toEqual( [
+			"introductionKeyword",
+			"metaDescriptionKeyword",
+			"metaDescriptionLength",
+			"textLength",
+			"titleKeyword",
+			"titleWidth"
+		] );
+	} );
+
+	it( "runs assessments that only require a text", function() {
 		assessor.assess( new Paper( "text" ) );
-		expect( assessor.getValidResults().length ).toBe( 7 );
-	});
+		let assessments = getResults( assessor.getValidResults() );
 
-	it( "additionally runs assessments that only require a keyword", function() {
-		assessor.assess( new Paper( "text", { keyword: "keyword" } ) );
-		expect( assessor.getValidResults().length ).toBe( 8 );
-	});
+		expect( assessments ).toEqual( [
+			"keyphraseLength",
+			"metaDescriptionLength",
+			"textImages",
+			"textLength",
+			"externalLinks",
+			"internalLinks",
+			"titleWidth"
+		] );
+	} );
 
-	it( "additionally runs assessments that require text and a keyword", function() {
+	it( "additionally runs assessments that require a keyword", function() {
 		assessor.assess( new Paper( "text", { keyword: "keyword" } ) );
-		expect( assessor.getValidResults().length ).toBe( 8 );
-	});
-
-	it( "additionally runs assessments that require an url", function() {
-		assessor.assess( new Paper( "text", { url: "sample url" } ) );
-		expect( assessor.getValidResults().length ).toBe( 7 );
-	});
-});
+		let assessments = getResults( assessor.getValidResults() );
+
+		expect( assessments ).toEqual( [
+			"introductionKeyword",
+			"metaDescriptionKeyword",
+			"metaDescriptionLength",
+			"textImages",
+			"textLength",
+			"externalLinks",
+			"internalLinks",
+			"titleKeyword",
+			"titleWidth"
+		] );
+	} );
+
+	it( "additionally runs assessments that require a keyword with a stopword ", function() {
+		assessor.assess( new Paper( "text", { keyword: "the keyword" } ) );
+		let assessments = getResults( assessor.getValidResults() );
+
+		expect( assessments ).toEqual( [
+			"introductionKeyword",
+			"keywordStopWords",
+			"metaDescriptionKeyword",
+			"metaDescriptionLength",
+			"textImages",
+			"textLength",
+			"externalLinks",
+			"internalLinks",
+			"titleKeyword",
+			"titleWidth"
+		] );
+	} );
+
+	it( "additionally runs assessments that require a url that is too long", function() {
+		assessor.assess( new Paper( "text", { url: "12345678901234567890123456789012345678901" } ) );
```

Comment:
I've seen some tests in this spec file before. Maybe we should find a way to prevent 'duplicate' tests?

Reply:
We've discussed the overlaps in the `spec/SEOAssessorSpec.js` and `spec/cornerstone/SEOAssessorSpec.js` files and decided that for now they can stay the way they are. However, we'll have to re-think the current approach once we e.g. add a new assessor.